multi-layer.md

```diff
@@ -0,0 +1,130 @@
+
+## Non-goals
+* This document doesn't provide definitions for any specific layer types.
+* Particular details of implementation will be provided in a different [document](layers-core.md)
+
+
+## Proposed changes
+
+Current WebXR spec is written with layers in mind (see [XRLayer interface](https://immersive-web.github.io/webxr/#xrlayer-interface)), however, only a single layer ([baseLayer](https://immersive-web.github.io/webxr/#dom-xrrenderstate-baselayer)) is currently supported.
+
+To support multiple layers the following changes are proposed:
+* Make the [XRRenderStateInit and XRRenderState](https://immersive-web.github.io/webxr/#xrrenderstate-interface)'s [baseLayer](https://immersive-web.github.io/webxr/#dom-xrrenderstate-baselayer) optional; add another optional member which will be the sequence of [XRLayer](https://immersive-web.github.io/webxr/#xrlayer-interface)s (let say we call it `layers`). Only one of the members should be used, either `baseLayer` or `layer`; if both are specified then `layers` takes precedence over the `baseLayer` (if supported).
+* The maximum number of supported layers could be requested from [XRSession](https://immersive-web.github.io/webxr/#xrsession) by using `getMaxLayerCount()` method; if the `getMaxLayerCount()` returns 0 then only the `baseLayer` is supported; if `getMaxLayerCount()` returns non-zero positive value, then `layers` can be used; 
+* Only `XRWebGLLayer` can be set to the `baseLayer`; only new layer types (`XRLayerProjection`, `XRLayerQuad`, etc) can be set as elements of `layers`.
+* The types of supported layers could be asked from [XRSession](https://immersive-web.github.io/webxr/#xrsession) by using `isLayerSupported(layer_class_name)` method;
+* Layers are drawn in the same order as they are specified in via `XRSession/updateRenderState`, with the 0th layer drawn first. Layers are drawn with a "painter’s algorithm," with each successive layer potentially overwriting the destination layers whether or not the new layers are virtually closer to the viewer. There is no depth testing between the layers. Layer's visibility is controlled by presence (or absence) of the layer in the sequence. To hide or show a layer - call [updateRenderState](https://immersive-web.github.io/webxr/#dom-xrsession-updaterenderstate) and provide a new sequence of layers.
+
+* Add common properties to [XRLayer](https://immersive-web.github.io/webxr/#xrlayer-interface) (such as chromatic aberration or alpha blending flags):
+* * `chromaticAberrationCorrection` - boolean, enables optical chromatic aberration correction for the layer when not done by default; chromatic aberration correction improves visual fidelity of the layer's rendering, but it might be computationally expensive; this flag may be ignored by the browser, if this feature is not supported.
```

Comment:
This seems rather low-level - on what basis should an application decide if chromatic aberration correction should be on or off? I think there's a fairly specific range of hardware performance where the platform has sufficient GPU resources to do chromatic aberration correction, but not enough to do so unconditionally for all layers. 

If you do want to keep this as a hint, please clarify that implementations are free to ignore the "correction off" value if it's not a performance concern, i.e. for a desktop VR system. As written, it could be interpreted as meaning that an implementation that supports chromatic aberration correction would be required to respect this flag, and that's likely not intended.

Bigger picture, is there a need for a more general performance hint system, i.e. providing optimization hints ("optimize this layer for best pixel quality") or feedback from the system on per-layer cost for certain feature sets? For example, a platform might support multiple layers, but performance would degrade if there's too many of them or if there's too much screen area overlap.

(This isn't intended to derail the discussion, but I think it's worth keeping an eye on the risk of apps tuning experiences based on user agent sniffing or similar due to lack of more direct feedback mechanisms.)